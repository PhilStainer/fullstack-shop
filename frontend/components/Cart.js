import {useQuery, useMutation} from '@apollo/react-hooks'
import {AiOutlineShopping} from 'react-icons/ai'

import ME_QUERY from '#root/graphql/me.query'
import {TOGGLE_CART_MUTATION} from '#root/graphql/localResolvers'

import StyledCart from '#root/components/styles/StyledCart'

const Cart = () => {
  const {data, loading, error} = useQuery(ME_QUERY)
  const [toggleCart] = useMutation(TOGGLE_CART_MUTATION)

  if (loading) return <div>Loading cart...</div>
  if (error) return <div>Failed to load cart...</div>

  const count = data.me.cart.reduce((total, item) => total + item.quantity, 0)

  return (
    <StyledCart title="Open cart" onClick={toggleCart} data-testid="cart">
      <AiOutlineShopping size="2rem" />
      <span data-testid="count">{count}</span>
    </StyledCart>
  )
}

export default Cart
