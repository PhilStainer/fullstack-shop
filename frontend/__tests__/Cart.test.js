import {render} from '@testing-library/react'

import Cart from '../components/Cart'

test('renders', () => {
  const {asFragment} = render(<Cart />)

  expect(asFragment()).toMatchSnapshot()
})
