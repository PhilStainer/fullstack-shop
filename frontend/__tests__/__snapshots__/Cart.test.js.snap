// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div>
    Â£2.00
  </div>
</DocumentFragment>
`;
